extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -1110,6 +1111,58 @@ export const popupMessageListener = (request: Request) => {
     return {};
   };
 
+  const storeSep24Data = async () => {
+    const {
+      publicKey,
+      txId,
+      sep10Url,
+      sep24Url,
+      status,
+      anchorDomain,
+      asset,
+    } = request.sep24Data;
+
+    await dataStorageAccess.setItem(
+      `${SEP24_DATA}:${publicKey}`,
+      JSON.stringify({
+        publicKey,
+        txId,
+        sep10Url,
+        sep24Url,
+        status,
+        anchorDomain,
+        asset,
+      }),
+    );
+  };
+
+  const loadSep24Data = async () => {
+    const publicKey = publicKeySelector(store.getState());
+
+    const res = JSON.parse(
+      (await dataStorageAccess.getItem(`${SEP24_DATA}:${publicKey}`)) || "{}",
+    );
+    return { sep24Data: res };
+  };
+
+  const storeSep24Status = async () => {
+    const { status } = request;
+    const publicKey = publicKeySelector(store.getState());
+    const res = JSON.parse(
+      (await dataStorageAccess.getItem(`${SEP24_DATA}:${publicKey}`)) || "{}",
```

Comment:
related to my comment above, after merging in 3.0.0, you'll be able to (and probably should) just set/get things in storage as their actual data type rather than JSON.stringifying before and JSON.parsing after